editor/plugins/material_editor_plugin.cpp

```diff
@@ -39,24 +39,16 @@
 
 void MaterialEditor::_notification(int p_what) {
 	switch (p_what) {
-		case NOTIFICATION_READY: {
-			//get_scene()->connect("node_removed",this,"_node_removed");
-
-			if (first_enter) {
-				//it's in propertyeditor so.. could be moved around
-
-				light_1_switch->set_normal_texture(get_theme_icon(SNAME("MaterialPreviewLight1"), SNAME("EditorIcons")));
-				light_1_switch->set_pressed_texture(get_theme_icon(SNAME("MaterialPreviewLight1Off"), SNAME("EditorIcons")));
-				light_2_switch->set_normal_texture(get_theme_icon(SNAME("MaterialPreviewLight2"), SNAME("EditorIcons")));
-				light_2_switch->set_pressed_texture(get_theme_icon(SNAME("MaterialPreviewLight2Off"), SNAME("EditorIcons")));
-
-				sphere_switch->set_normal_texture(get_theme_icon(SNAME("MaterialPreviewSphereOff"), SNAME("EditorIcons")));
-				sphere_switch->set_pressed_texture(get_theme_icon(SNAME("MaterialPreviewSphere"), SNAME("EditorIcons")));
-				box_switch->set_normal_texture(get_theme_icon(SNAME("MaterialPreviewCubeOff"), SNAME("EditorIcons")));
-				box_switch->set_pressed_texture(get_theme_icon(SNAME("MaterialPreviewCube"), SNAME("EditorIcons")));
-
-				first_enter = false;
-			}
+		case NOTIFICATION_THEME_CHANGED: {
```

Comment:
Does `THEME_CHANGED` get called also when instantiating a new editor plugin? I think usually we make such logic run both for `READY` and `THEME_CHANGED` to handle both, no?

Reply:
ah, I didn't remember really about this part of the engine - I will return a NOTIFICATION_ENTER_TREE instead. But when you simply runned an editor its calling THEME_CHANGED which is strange a bit.

Reply:
@akien-mga I don't think it's necessary, see https://github.com/godotengine/godot/issues/50743. I made an experimental [commit](https://github.com/AaronRecord/godot/commit/960fd97f8a42b9f62ef501717095ef6d81286e1b) where I removed notification enter tree, and one of the test cases fails, but the editor starts and closes and seems to run fine. I'm hoping to clean up the theme propagation logic but I'm currently away from home for a few days so I can't work on it.